src/Escape.elm

```diff
@@ -0,0 +1,55 @@
+module Escape exposing (SanitizeLevel(..), sanitize)
+
+import Regex exposing (Regex)
+
+
+type alias Replacement =
+    String
+
+
+type SanitizeLevel
+    = Namespace
+    | Element
+    | State
+
+
+sanitizeRegex : Regex
+sanitizeRegex =
+    Regex.regex "[^a-z0-9\\-_]"
+        |> Regex.caseInsensitive
+
+
+sanitize : SanitizeLevel -> String -> String
+sanitize level str =
+    case level of
+        Namespace ->
+            sanitize_ sanitizeRegex "" str
+
+        Element ->
+            sanitize_ sanitizeRegex "" str
+                |> removeDoubleUnderscore
+
+        State ->
+            sanitize_ sanitizeRegex "" str
+                |> removeDoubleUnderscore
```

Comment:
here it leads to unnecessary code duplication. With function composition this will be much cleaner.